### PR TITLE
feat: added support for AWS LocalZone Public and Private Subnets

### LINKED ISSUE #187: [FEATURE]Add LocalZone support for VPC module
**Describe the solution you'd like**
Need support for creating Public and Private subnets in AWS LocalZone 

**Additional context**

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*
Fixes #187 

*Description of changes:*
added support for AWS LocalZone Public and Private Subnets

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: cc6e77fac79844d72651acead0f7c7831babd194
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=1bbd2e40-3512-41d0-a53c-ff73ef257198%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 5dd4a5f60b5fcee1724d31a8f11c68f691cfd757
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4cd15163-d504-4f95-a11f-b9ad7693e5f6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 7bc5eb97d8ad82046facd1ff9c3dacccdf8085f9
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=94922ead-e656-4fd6-a881-a20aac325fd4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: a395a4989a5d04f91ef8adbd55417dc840d71b9b
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cd35fafb-eade-4b4c-9ff6-bb81858085c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: d21443ced7d833ed4bba9c498b5a8dfac0e74ad1
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=77c78fa5-1aca-4894-8bca-031d5f57a0f5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 17d242210568a4e64970785135dd0bb4d17db4d0
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=516e9a38-0110-472a-a3ea-f86d41beb6ea%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by kevinsoucy** [comment]
looks good to me!

File: modules/network/basic-cdk/stack.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 
 _logger: logging.Logger = logging.getLogger(__name__)
 
+VPC_CIDR = "10.0.0.0/16"
+SUBNET_SIZE = 24
```
Comment (kukushking) [inline modules/network/basic-cdk/stack.py:16]: Shall we add input parameters for these then?

Comment (srinivasreddych) [reply modules/network/basic-cdk/stack.py:16]: I thought about it, but at this point of time or even in the future, i see only us using the networking module to deploy the required networking. Ideally customers will use other mechanisms such as LZs etc

Comment (kukushking) [reply modules/network/basic-cdk/stack.py:16]: Ok makes sense

**Review by dgraeber — DISMISSED** [review]

**Review by kukushking — DISMISSED** [review]
Looks good!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 0c5cf164cdc3109f5eb004902b74a06578c6302d
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d4223d69-d81e-4e7a-962f-5368c66d2b9c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Thanks!

**Pull request merged**